src/pages/index.js

```diff
@@ -37,28 +44,61 @@ cardImagePopup.setEventListeners();
 function handleCardFormSubmit(inputValues) {
   const name = inputValues.title;
   const link = inputValues.link;
-  renderCard({ name, link });
+  // renderCard({ name, link });
+  addCardPopup.setLoadingState(true);
+  api.createNewCard(name, link)
+    .then((data) => {
+      renderCard(data);
+    })
+    .catch((err) => {
+      console.log(err);
+    }).finally(() => {
+      addCardPopup.setLoadingState(false);
+    })
   addCardPopup.close();
   addCardForm.reset();
   addCardFormValidator.toggleButtonState();
 }
-
+const handleCardLike = (cardId, cardInstance) => {
+  const isItLiked = cardInstance.like;
+  console.log(cardInstance.like);
+  const toggleLike = isItLiked ? api.likeButtonDeactive(cardId) : api.likeButtonActive(cardId);
+  toggleLike
+    .then((updatedCardData) => {
+      cardInstance.handleLike(!isItLiked);
+    })
+    .catch((err) => {
+      console.error("like toogle error:", err)
+    })
+}
 function createCard(cardData) {
   const cardElements = new Card(cardData, "#card-template", () => {
-    cardImagePopup.open(cardData);
-  });
+    cardImagePopup.open(cardData)
+  },
+    (cardId) => {
+      deleteCardPopup.open(cardId);
+
+    }, handleCardLike,
+  );
+
   return cardElements.viewCard();
-}
+};
 
 function renderCard(cardData) {
   const newCard = createCard(cardData);
+  newCard.id = cardData._id;
   cardSection.addItem(newCard);
 }
 
 
 /*                                                                          */
 /*                                EVENT LISTENERS                           */
 /*                                                                          */
+/*change profile picture*/
+editProfileImageIcon.addEventListener("click", () => {
+  changeProfilePhoto.open();
+  changeProfilePhoto.setEventListener();
```

Comment:
`.setEventListener();`  should be removed from `opening` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This may cause a **`memory leak`**